src/main/scala/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.scala

```diff
@@ -521,14 +596,18 @@ class CollectDuplexSeqMetrics
 
   /** Writes out all the metrics and plots. */
   private[umi] def write(description: String): Unit = {
-    val Seq(fsPath, dfsPath, umiPath, yieldPath, pdfPath) = Seq(FamilySizeMetricsExt, DuplexFamilySizeMetricsExt, UmiMetricsExt, YieldMetricsExt, PlotsExt).map { ext =>
-      output.getParent.resolve(output.getFileName + ext)
-    }
+    val Seq(fsPath, dfsPath, umiPath, duplexUmiPath, yieldPath, pdfPath) =
+      Seq(FamilySizeMetricsExt, DuplexFamilySizeMetricsExt, UmiMetricsExt, DuplexUmiMetricsExt, YieldMetricsExt, PlotsExt).map { ext =>
+        output.getParent.resolve(output.getFileName + ext)
+      }
 
     Metric.write(fsPath, familySizeMetrics)
     Metric.write(dfsPath, duplexFamilySizeMetrics)
     Metric.write(yieldPath, yieldMetrics)
-    Metric.write(umiPath, umiMetrics)
+
+    val umiMetricValues = umiMetrics
```

Comment:
I have it set up so that all the functions like `umiMetrics` are functions that calculate the metrics each time they are called (so, e.g. they can be sampled many times during testing).  Here I'm just caching the values that are generated, since I need them twice.